globus_automate_client/cli/constants.py

```diff
@@ -24,10 +23,18 @@ class FlowRoleDeprecated(str, enum.Enum):
     administered_by = "administered_by"
 
 
-# Adapted from https://stackoverflow.com/questions/33679930/how-to-extend-python-enum
-FlowRoleAllNames = enum.Enum(
-    "FlowRoleAllNames", [(i.name, i.value) for i in chain(FlowRole, FlowRoleDeprecated)]
-)
+class FlowRoleAllNames(str, enum.Enum):
```

Comment:
I wonder if this could be left as the original version, and we use this:
```python
from typing import TypeVar, Union

T = TypeVar("T", bound=Union[FlowRole, FlowRoleDeprecated, FlowRoleAllNames])
```
That might work? Is mypy appeased?

Reply:
It can't be left as-is -- mypy's complaint is not the types but rather that dynamic enums fundamentally break the concept of having enums.

Reply:
I'm marking this as resolved. Please re-open if this needs to be addressed differently.